tool/provision_integration_test_resources.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+set -e
+
+if [[ -z "$1" ]]; then
+    echo "Usage: provision_integration_test_resources.sh <CATEGORY>" >&2
+    echo "<CATEGORY> = analytics, auth, storage" >&2
+    exit 1
+fi
+
+pushd infra
+pnpm install
```

Comment:
question: What's the rationale for using pnpm instead of npm/yarn here? Great tool but we already have to use other build systems for other repos like docs so I just want to understand the rationale for introducing another one here.

Reply:
It makes a significant difference in the size of node_modules since it uses a central cache. I think there are other tools which do this now but this one came highly recommended as a replacement for npm.